cli/src/pcluster/models/cluster_resources.py

```diff
@@ -74,6 +74,22 @@ def batch_compute_environment(self):
         """Return Batch compute environment."""
         return self._get_output("BatchComputeEnvironmentArn")
 
+    def get_failure_reason(self):
+        """Reason of the failure when the cluster_status is in CREATE_FAILED."""
+        stack_events = get_all_stack_events(self.name)
```

Comment:
You can use https://docs.python.org/3/library/itertools.html#itertools.chain to avoid dealing with the chunks by lazily flattening the result at this call site

Reply:
👍 lesson learned